ecommerce/theming/helpers.py

```
@@ -2,119 +2,113 @@
+        return None
 
-    # Theming is enabled
-    return True
+    site_theme = get_current_site_theme()
+    if not site_theme:
+        return None
+    try:
+        return Theme(
+            name=site_theme.theme_dir_name,
+            theme_dir_name=site_theme.theme_dir_name,
+            themes_base_dir=get_theme_base_dir(site_theme.theme_dir_name),
+        )
+    except ValueError as e:
+        # Log exception message and return None, so that open source theme is used instead
+        logger.exception('Theme not found in any of the themes dirs. [%s]', e)
+        return None
 
 
-def get_site_theme_cache_key(site):
+def get_theme_base_dir(theme_dir_name, suppress_error=False):
     """
-    Return cache key for the given site.
-
-    Example:
-        >> site = Site(domain='red-theme.org', name='Red Theme')
-        >> get_site_theme_cache_key(site)
-        'theming.site.red-theme.org'
+    Returns absolute path to the directory that contains the given theme.
 
-    Parameters:
-        site (django.contrib.sites.models.Site): site where key needs to generated
+    Args:
+        theme_dir_name (str): theme directory name to get base path for
+        suppress_error (bool): if True function will return None if theme is not found instead of raising an error
     Returns:
-        (str): a key to be used as cache key
-    """
-    cache_key = "theming.site.{domain}".format(
-        domain=site.domain
-    )
-    return cache_key
-
-
-def cache_site_theme_dir(site, theme_dir):
+        (str): Base directory that contains the given theme
     """
-    Cache site's theme directory.
+    for themes_dir in get_theme_base_dirs():
+        if theme_dir_name in (_dir for _dir in os.listdir(themes_dir) if is_theme_dir(themes_dir / _dir)):
+            return themes_dir
 
-    Example:
-        >> site = Site(domain='red-theme.org', name='Red Theme')
-        >> cache_site_theme_dir(site, 'red-theme')
+    if suppress_error:
```

Does it make sense to write a log entry here if we are suppressing the error?  I don't want to create additional noise if it's not useful information.